internal/service/meter.go

```diff
@@ -63,3 +64,27 @@ func (s *meterService) DisableMeter(ctx context.Context, id string) error {
 	}
 	return s.meterRepo.DisableMeter(ctx, id)
 }
+
+func (s *meterService) UpdateMeter(ctx context.Context, id string, filters []meter.Filter) (*meter.Meter, error) {
+	if id == "" {
+		return nil, fmt.Errorf("id is required")
+	}
+
+	if len(filters) == 0 {
+		return nil, fmt.Errorf("filters cannot be empty")
+	}
+
+	// Call the repository to update filters
+	err := s.meterRepo.UpdateMeter(ctx, id, filters)
+	if err != nil {
+		return nil, fmt.Errorf("update meter: %w", err)
+	}
+
+	// Fetch the updated meter
+	updatedMeter, err := s.meterRepo.GetMeter(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("get meter: %w", err)
+	}
+
+	return updatedMeter, nil
```

Comment:
don't fetch right after writing updating the item in DB
return the edited object